### PR TITLE
Add content of the draft of our browser support strategy to our docs

### PR DESCRIPTION
**Pull request by romaricpascal**
> **WIP** The README still needs updating following [this discussion on Slack (internal link)](https://gds.slack.com/archives/C05P6FJ1YH2/p1698048065711409) to link to the docs

Adds the content of [the draft of the full version of our browser support strategy to the docs (internal link)](https://docs.google.com/document/d/1U5yQPYkomtPsLx3rPoNRrF9Kbei8KsIfnOSEUPMfbpM/edit?usp=sharing).

The location is based on [this discussion on Slack (internal link)](https://gds.slack.com/archives/C05P6FJ1YH2/p1698048065711409).

I've brought the range of Safari versions [from the summary](https://github.com/alphagov/govuk-frontend-docs/pull/356).

The table has to be in HTML due to some complex content in the cells. I've opted for semantics when marking up the list of browsers in there, which adds bullets at the start of the list items. If we prefer the formatting from the document, we can go with paragraphs and `<br>`:

```html
<p>
Chrome 61+<br>
Edge 16-18<br>
Edge 79+<br>
Safari 11+ (mac)<br>
Firefox 60+<br>
Opera 48+
</p>
<p>
Safari 10.3+ (iOS)<br>
Samsung Internet 8.2+
</p>
```

Part of alphagov/govuk-frontend-docs#340 (the issue needs to wait for [the guidance summary to have been merged on the `govuk-frontend-docs` repo](https://github.com/alphagov/govuk-frontend-docs/pull/356) before being closed).

### CODE REVIEW

File: docs/contributing/browser-support.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
+# How we provide support for different browsers
+
+Users can access government services through a wide range of new to older browsers. Our components provide a baseline experience through HTML and CSS, allowing all users to complete their tasks even [if JavaScript is not working](https://www.kryogenix.org/code/browser/everyonehasjs.html). Some components have JavaScript enhancements which add extra features or fix accessibility issues.
```
Comment (claireashworth) [inline docs/contributing/browser-support.md:3]:
```suggestion
Users can access government services through a wide range of new to older browsers. Our components provide a baseline experience through HTML and CSS, allowing all users to complete their tasks even if JavaScript is not working. Some components have JavaScript enhancements which add extra features or fix accessibility issues.
```

File: docs/contributing/browser-support.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+- necessary enhancements being available to users
+- optional enhancements being available to users
+- bug prioritisation
+
+This table shows the level of support for the 3 grades of browsers (A, B and C) which run our JavaScript.
+
+We’ve excluded Grade X browsers from this table as we’re not supporting enhancements for them and the prioritisation for bugs is ‘will not fix’.
+
+<table>
+    <tbody>
```
Comment (colinrotherham) [inline docs/contributing/browser-support.md:71]: The table looks great on wider screens, but it's super squished on mobiles 😣

Think we can find an alternative?

Comment (colinrotherham) [reply docs/contributing/browser-support.md:71]: We disable Prettier `embeddedLanguageFormatting` but probably stick to 2-spaces?

```suggestion
<table>
  <tbody>
```

Comment (claireashworth) [reply docs/contributing/browser-support.md:71]: I can move everything out into bullet points for each grade?

Comment (domoscargin) [reply docs/contributing/browser-support.md:71]: Suggest we leave the table in for this pre-release, then think about another solution for the next.

**Review by domoscargin — APPROVED** [review]
Looks good!

**Pull request merged**